site/src/components/service-message-list.tsx

```diff
@@ -0,0 +1,187 @@
+import {
+  ServiceDefinition,
+  ServiceMessageDefinition,
+} from "@blockprotocol/core";
+import { Box, Typography } from "@mui/material";
+import * as React from "react";
+import slugify from "slugify";
+
+import { JsonSchema } from "../lib/json-schema";
+import { mdxComponents } from "../util/mdx-components";
+
+const DataType: React.VoidFunctionComponent<{ propertySchema: JsonSchema }> = ({
+  propertySchema,
+}) => {
+  const MdxA = mdxComponents.a!;
+
+  const isArray = propertySchema.type === "array";
+  const { type: innerType, $ref } = isArray
+    ? (propertySchema.items as JsonSchema)
+    : propertySchema;
+
+  return $ref ? (
+    <MdxA href={$ref} target="_blank">
+      {$ref.match(/[^/]+(?=\/$|$)/)?.[0]}
+      {isArray ? "[]" : ""}
+    </MdxA>
+  ) : (
+    <>
+      {innerType}
+      {isArray ? "[]" : ""}
+    </>
+  );
+};
+
+const ServiceMessageData: React.VoidFunctionComponent<{
+  data: ServiceMessageDefinition["data"];
+}> = ({ data }) => {
+  const MdxCode = mdxComponents.code!;
+  const MdxTable = mdxComponents.table!;
+
+  const { type } = data;
+
+  if (type === "object") {
+    return (
+      <>
+        <MdxCode>object</MdxCode>
+        <MdxTable>
+          <thead>
+            <tr>
+              <th>property</th>
+              <th>type</th>
+              <th>required</th>
+              <th>description</th>
+            </tr>
```

Comment:
Nit: appreciate the table styling was not introduced in this PR, but might be worth considering giving the table heading a grey background colour, to help distinguish it from the contents a bit more?
<img width="684" alt="image" src="https://user-images.githubusercontent.com/42802102/174757027-cbd48f61-b506-4b70-82d7-8d3b05c645f9.png">

Reply:
I've added some grey in https://github.com/blockprotocol/blockprotocol/pull/387/commits/3e8b7528d05f717172be277966b30038d4694135, and also fixed the misplaced `required` which your screenshot reveals – thanks!

Reply:
Nice, thanks Ciaran 🙂